x-pack/plugins/canvas/canvas_plugin_src/functions/common/if.ts

```diff
@@ -38,12 +42,8 @@ export function ifFn(): ExpressionFunctionDefinition<'if', unknown, Arguments, u
         help: argHelp.else!,
       },
     },
-    fn: async (input, args) => {
-      if (args.condition) {
-        return args.then?.().pipe(take(1)).toPromise() ?? input;
-      } else {
-        return args.else?.().pipe(take(1)).toPromise() ?? input;
-      }
+    fn(input, args) {
+      return defer(() => (args.condition ? args.then?.() : args.else?.()) ?? of(input));
```

Comment:
condition is `required`

Reply:
Corrected the argument type, but the code remains the same because `then` and `else` are optional.